### PR TITLE
zoom: Add a different `pre_install` script for each architecture

### LINKED ISSUE #9482: zoom@5.12.2.9281: decompress error
#### Console Output
```
❯ scoop install zoom
Installing 'zoom' (5.12.2.9281) [64bit] from extras bucket
ZoomInstallerFull.exe (56.2 MB) [============================================================================] 100%
Checking hash of ZoomInstallerFull.exe ... ok.
Extracting dl.7z ... done.
Running pre_install script...
ERROR Exit code was 2!
Failed to extract files from C:\Users\User\scoop\apps\zoom\5.12.2.9281\Zoom.msi.
Log file:
  ~\scoop\apps\zoom\5.12.2.9281\7zip.log

Please try again or create a new issue by using the following link and paste your console output:
https://github.com/ScoopInstaller/Extras/issues/new?title=zoom%405.12.2.9281%3a+decompress+error
```

#### Logfile
```
❯ cat C:\Users\User\scoop\apps\zoom\5.12.2.9281\7zip.log

7-Zip 22.01 (x64) : Copyright (c) 1999-2022 Igor Pavlov : 2022-07-15

Scanning the drive for archives:

ERROR: ("The specified file cannot be found." in my language)
C:\Users\User\scoop\apps\zoom\5.12.2.9281\Zoom.msi



System ERROR:
("The specified file cannot be found." in my language)
```

#### Directory
```
❯ ls C:\Users\User\scoop\apps\zoom\5.12.2.9281\

    Directory: C:\Users\User\scoop\apps\zoom\5.12.2.9281

Mode                 LastWriteTime         Length Name
----                 -------------         ------ ----
-a---          2022/10/15    22:07            306 7zip.log
-a---          2022/10/09    19:15         977320 Installer.exe
-a---          2022/10/09    19:11       58268211 ZoomFull_Sip.cab
```

#### Scoop Configuration

```json
{
  "last_update": "2022-10-15T21:39:52.6086482+09:00",
  "scoop_repo": "https://github.com/ScoopInstaller/Scoop",
  "scoop_branch": "master"
}
```

### LINKED ISSUE #9486: zoom@5.12.2.9281: decompress error
#### Scoop Configuration

```json
{
  "last_update": "2022-10-15T17:37:16.4467466-03:00",
  "scoop_repo": "https://github.com/ScoopInstaller/Scoop",
  "scoop_branch": "master"
}
```

### LINKED ISSUE #9487: zoom@5.12.2.9281: decompress error
#### Console Output
```
zoom: 5.11.4.7185 -> 5.12.2.9281
Updating one outdated app:
Updating 'zoom' (5.11.4.7185 -> 5.12.2.9281)
Downloading new version
Starting download with aria2 ...
Download: Download Results:
Download: gid   |stat|avg speed  |path/URI
Download: ======+====+===========+=======================================================
Download: 886295|OK  |    14MiB/s|C:/Users/user/scoop/cache/zoom#5.12.2.9281#https_cdn.zoom.us_prod_5.12.2.9281_x64_ZoomInstallerFull.exe_dl.7z
Download: Status Legend:
Download: (OK):download completed.
Checking hash of ZoomInstallerFull.exe ... ok.
Uninstalling 'zoom' (5.11.4.7185)
Running uninstaller script...
Unregistering zoommtg:// protocol handler
Removing shim 'Zoom.shim'.
Removing shim 'Zoom.exe'.
Unlinking ~\scoop\apps\zoom\current
Installing 'zoom' (5.12.2.9281) [64bit] from extras bucket
Loading ZoomInstallerFull.exe from cache.
Extracting dl.7z ... done.
Running pre_install script...
ERROR Exit code was 2!
Failed to extract files from C:\Users\user\scoop\apps\zoom\5.12.2.9281\Zoom.msi.
Log file:
  ~\scoop\apps\zoom\5.12.2.9281\7zip.log
```
#### scoop\apps\zoom\5.12.2.9281\7zip.log
```
7-Zip 22.01 (x64) : Copyright (c) 1999-2022 Igor Pavlov : 2022-07-15

Scanning the drive for archives:

ERROR: The system cannot find the file specified.
C:\Users\user\scoop\apps\zoom\5.12.2.9281\Zoom.msi



System ERROR:
The system cannot find the file specified.
```

#### Scoop Configuration
```json
{
  "aria2-enabled": true,
  "aria2-warning-enabled": false,
  "aria2-max-connection-per-server": "10",
  "last_update": "2022-10-15T20:15:36.8670900-04:00",
  "scoop_repo": "https://github.com/ScoopInstaller/Scoop",
  "scoop_branch": "master"
}
```

### PR DESCRIPTION
**Pull request by Zliced13**
Closes #9487, Closes #9486, and Closes #9482
Relates to #9485

- [x] I have read the [Contributing Guide](https://github.com/ScoopInstaller/.github/blob/main/.github/CONTRIBUTING.md).

### CODE REVIEW

**Comment by Zliced13** [comment]
@niiheaven

File: bucket/zoom.json [hunk 1]
```diff
@@ -9,21 +9,20 @@
...
             "url": "https://cdn.zoom.us/prod/5.12.2.9281/ZoomInstaller.exe#/dl.7z",
-            "hash": "02a6d6c34b332fedbcd0efad99d36e7c5570932bfaff575f95443f38cbdb31d8"
+            "hash": "02a6d6c34b332fedbcd0efad99d36e7c5570932bfaff575f95443f38cbdb31d8",
+            "pre_install": "Remove-Item \"$dir\\Installer.exe\"; Expand-7zipArchive \"$dir\\Zoom.msi\" -Removal"
         },
         "arm64": {
             "url": "https://cdn.zoom.us/prod/5.12.2.9281/arm64/ZoomInstallerFull.exe#/dl.7z",
-            "hash": "4469285d3ecaa03b2bc644f3beb755ce0b8530327ca24790a733f78117fa54f8"
+            "hash": "4469285d3ecaa03b2bc644f3beb755ce0b8530327ca24790a733f78117fa54f8",
+            "pre_install": "Remove-Item \"$dir\\Installer.exe\"; Expand-7zipArchive \"$dir\\ZoomFull_Sip.cab\" -Removal"
```
Comment (HUMORCE) [inline bucket/zoom.json:23]:
```suggestion
            "pre_install": "Expand-7zipArchive \"$dir\\ZoomFull_Sip.cab\" -Removal"
```

File: bucket/zoom.json [hunk 2]
```diff
@@ -9,21 +9,20 @@
...
             "url": "https://cdn.zoom.us/prod/5.12.2.9281/x64/ZoomInstallerFull.exe#/dl.7z",
-            "hash": "f7cba2e790a7d41658ea0f430ac775f08d9872add59168fd72f12262bc9f9786"
+            "hash": "f7cba2e790a7d41658ea0f430ac775f08d9872add59168fd72f12262bc9f9786",
+            "pre_install": "Remove-Item \"$dir\\Installer.exe\"; Expand-7zipArchive \"$dir\\ZoomFull_Sip.cab\" -Removal"
         },
         "32bit": {
             "url": "https://cdn.zoom.us/prod/5.12.2.9281/ZoomInstaller.exe#/dl.7z",
-            "hash": "02a6d6c34b332fedbcd0efad99d36e7c5570932bfaff575f95443f38cbdb31d8"
+            "hash": "02a6d6c34b332fedbcd0efad99d36e7c5570932bfaff575f95443f38cbdb31d8",
+            "pre_install": "Remove-Item \"$dir\\Installer.exe\"; Expand-7zipArchive \"$dir\\Zoom.msi\" -Removal"
```
Comment (HUMORCE) [inline bucket/zoom.json:18]:
```suggestion
            "pre_install": "Expand-7zipArchive \"$dir\\Zoom.msi\" -Removal"
```

File: bucket/zoom.json [hunk 3]
```diff
@@ -9,21 +9,20 @@
     "architecture": {
         "64bit": {
             "url": "https://cdn.zoom.us/prod/5.12.2.9281/x64/ZoomInstallerFull.exe#/dl.7z",
-            "hash": "f7cba2e790a7d41658ea0f430ac775f08d9872add59168fd72f12262bc9f9786"
+            "hash": "f7cba2e790a7d41658ea0f430ac775f08d9872add59168fd72f12262bc9f9786",
+            "pre_install": "Remove-Item \"$dir\\Installer.exe\"; Expand-7zipArchive \"$dir\\ZoomFull_Sip.cab\" -Removal"
```
Comment (HUMORCE) [inline bucket/zoom.json:13]:
```suggestion
            "pre_install": "Expand-7zipArchive \"$dir\\ZoomFull_Sip.cab\" -Removal"
```

File: bucket/zoom.json [hunk 4]
```diff
@@ -9,21 +9,20 @@
...
+            "hash": "4469285d3ecaa03b2bc644f3beb755ce0b8530327ca24790a733f78117fa54f8",
+            "pre_install": "Remove-Item \"$dir\\Installer.exe\"; Expand-7zipArchive \"$dir\\ZoomFull_Sip.cab\" -Removal"
         }
     },
-    "pre_install": [
-        "Expand-7zipArchive \"$dir\\Zoom.msi\" -Removal",
-        "Remove-Item \"$dir\\Install*\""
-    ],
     "post_install": [
         "Write-Host \"Registering zoommtg:// protocol handler\"",
```
Comment (HUMORCE) [inline bucket/zoom.json:27]:
```suggestion
        "Remove-Item \"$dir\\Install*\"",
        "Write-Host \"Registering zoommtg:// protocol handler\"",
```

**Review by HUMORCE — CHANGES_REQUESTED** [review]
duplicate lines

File: bucket/zoom.json (hunk 3 above)
Comment (Zliced13) [reply bucket/zoom.json:13]: Why should I **not** remove the installer.exe file?

**Comment by Zliced13** [comment]
I added your suggestions to the manifest.

**Review by HUMORCE — APPROVED** [review]

**Pull request merged**